### ISSUE TITLE
[TypeScript] Fetcher function doesn't respect array of keys types

### ISSUE THREAD

**Issue by enzoferey** [vercel/swr#331]
With a piece of code like:

```ts
const key = `some graphql query with variables`;
const entityId = 1;

const fetcher = (key: string, entityId: number) => { ... }

const { data } = useSWR([key, entityId], fetcher);
```

If you now inverse the arguments of `fetcher`, TypeScript doesn't report a type error even though it would be receiving the wrong parameters in runtime.

**Comment by crutch12** [comment]
Bump
Any progress on this?

**Comment by shuding** [comment]
Fixed with #1477 🎉
